test-framework/sudo-compliance-tests/src/env_reset.rs

```diff
@@ -124,3 +105,152 @@ fn env_reset_mode_clears_env_vars() -> Result<()> {
+
+    assert_eq!(HashMap::new(), sudo_env);
+
+    Ok(())
+}
+
+#[test]
+fn some_vars_are_preserved() -> Result<()> {
+    let env = Env(SUDOERS_ROOT_ALL_NOPASSWD).build()?;
+
+    let sudo_abs_path = Command::new("which").arg("sudo").exec(&env)?.stdout()?;
+    let env_abs_path = Command::new("which").arg("env").exec(&env)?.stdout()?;
+
+    let home = "some-home";
+    let mail = "some-mail";
+    let shell = "some-shell";
+    let logname = "some-logname";
+    let user = "some-user";
+    let display = "some-display";
+    let path = "some-path";
+    let term = "some-term";
+    let sudo_command = "some-sudo-command";
+    let sudo_user = "some-sudo-user";
+    let sudo_uid = "some-sudo-uid";
+    let sudo_gid = "some-sudo-gid";
+    let stdout = Command::new("env")
+        .args([
+            "-i",
+            "SUDO_RS_IS_UNSTABLE=I accept that my system may break unexpectedly",
+            &format!("HOME={home}"),
+            &format!("MAIL={mail}"),
+            &format!("SHELL={shell}"),
+            &format!("LOGNAME={logname}"),
+            &format!("USER={user}"),
+            &format!("DISPLAY={display}"),
+            &format!("PATH={path}"),
+            &format!("TERM={term}"),
+            &format!("SUDO_COMMAND={sudo_command}"),
+            &format!("SUDO_USER={sudo_user}"),
+            &format!("SUDO_UID={sudo_uid}"),
+            &format!("SUDO_GID={sudo_gid}"),
+            &sudo_abs_path,
+            &env_abs_path,
+        ])
+        .exec(&env)?
+        .stdout()?;
+    let mut sudo_env = helpers::parse_env_output(&stdout)?;
+
+    // not preserved
+    assert_eq!(Some("/root"), sudo_env.remove("HOME"));
+    assert_eq!(Some("/var/mail/root"), sudo_env.remove("MAIL"));
+    assert_eq!(Some("/bin/bash"), sudo_env.remove("SHELL"));
+    assert_eq!(Some("root"), sudo_env.remove("LOGNAME"));
+    assert_eq!(Some("root"), sudo_env.remove("USER"));
+    assert_eq!(
+        Some(env_abs_path).as_deref(),
+        sudo_env.remove("SUDO_COMMAND")
+    );
+    assert_eq!(Some("root"), sudo_env.remove("SUDO_USER"));
+    assert_eq!(Some("0"), sudo_env.remove("SUDO_UID"));
+    assert_eq!(Some("0"), sudo_env.remove("SUDO_GID"));
+
+    // preserved
+    assert_eq!(Some(display), sudo_env.remove("DISPLAY"));
+    assert_eq!(Some(path), sudo_env.remove("PATH"));
+    assert_eq!(Some(term), sudo_env.remove("TERM"));
+
+    assert_eq!(HashMap::new(), sudo_env);
+
+    Ok(())
+}
+
+// only relevant to preserved env vars
+#[test]
+#[ignore]
+fn vars_whose_values_start_with_parentheses_are_removed() -> Result<()> {
+    let env = Env(SUDOERS_ROOT_ALL_NOPASSWD).build()?;
+
+    let sudo_abs_path = Command::new("which").arg("sudo").exec(&env)?.stdout()?;
+    let env_abs_path = Command::new("which").arg("env").exec(&env)?.stdout()?;
+
+    let stdout = Command::new("env")
+        .args([
+            "-i",
+            "SUDO_RS_IS_UNSTABLE=I accept that my system may break unexpectedly",
```

Comment:
I wonder if this should be in a constant

Reply:
that's a good idea. I'll do that refactor in a follow up PR.